infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -137,21 +145,36 @@ public void setupTopologyConfigId(long topologyConfigId) {
         long persistedTopologyConfigId = query(timestamp, LogReplicationMetadataType.TOPOLOGY_CONFIG_ID);
 
         if (topologyConfigId <= persistedTopologyConfigId) {
-            log.warn("Skip setupTopologyConfigId. the current topologyConfigId " + topologyConfigId + " is not larger than the persistedTopologyConfigID " + persistedTopologyConfigId);
+            log.warn("Skip setupTopologyConfigId. the current topologyConfigId {} is not larger than the persistedTopologyConfigID {}",
+                topologyConfigId, persistedTopologyConfigId);
             return;
         }
 
         TxBuilder txBuilder = corfuStore.tx(NAMESPACE);
 
-        for (LogReplicationMetadataType key : LogReplicationMetadataType.values()) {
-            long val = Address.NON_ADDRESS;
-            if (key == LogReplicationMetadataType.TOPOLOGY_CONFIG_ID) {
-                val = topologyConfigId;
+        for (LogReplicationMetadataType type : LogReplicationMetadataType.values()) {
+            if (type == LogReplicationMetadataType.TOPOLOGY_CONFIG_ID) {
+                appendUpdate(txBuilder, type, topologyConfigId);
+                try {
+                    IRetry.build(IntervalRetry.class, () -> {
```

Comment:
Will we add retry to each tx commit in metadata manager? I think we should have a max retry limit here.